internal/controller/runtime/fsm/runtime_fsm_configure_skr.go

```diff
@@ -35,25 +60,47 @@ func sFnConfigureOidc(ctx context.Context, m *fsm, s *systemState) (stateFn, *ct
 	err := recreateOpenIDConnectResources(ctx, m, s, additionalOIDCStatus)
 
 	if err != nil {
-		updateConditionFailed(&s.instance)
-		m.log.Error(err, "Failed to create OpenIDConnect resource. Scheduling for retry")
+		updateConditionFailed(&s.instance, imv1.ConditionReasonOidcError , oidcErrorMessage)
+		m.log.Error(err, oidcErrorMessage)
 		return requeue()
```

Comment:
The `requeue()` function will not update Status with the condition values set by `updateConditionFailed` . To store state before requeue you should use `updateStatusAndRequeue() `

Reply:
done in fdc444f09315c83849f6d950881a99e47f1d0101